### ISSUE TITLE
Can't log in with Azure Account - error: Unable to sign in: Method 'get_SerializationSettings' ...

### ISSUE THREAD

**Issue by thereforeattila** [microsoft/AzureStorageExplorer#2244]
**Storage Explorer Version**: 1.11.0
**Build Number**: 20191105.2
**Platform/OS**: Windows 10
**Architecture**: x64
**Regression From**:  Yes, it was working until a few versions ago I removed my account and tried to add it back.

## Bug Description ##
When I try to add an Azure Account (Environment: Azure) I always get the following error after authenticating (doesn't matter if it's interactive or device code flow sign-in):

Unable to sign in: Method 'get_SerializationSettings' in type 'Microsoft.Azure.Management.ResourceManager.SubscriptionClient' from assembly 'Microsoft.Azure.Management.ResourceManager, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.

## Steps to Reproduce ##
1. Try to log in.
2. Error will happen no matter what I do.

## Expected Experience ##
Login should work :)

## Actual Experience ##
Doesn't work.

## Additional Context ##
I think this may have to do something with the old AzureRM module but I've cleaned and uninstalled it already and it still doesn't work. Also all other authentication methods (SAS, Storage key, etc) work fine without an issue. It's only the Azure Account login which doesn't work.
![error](https://user-images.githubusercontent.com/57538696/68496596-0b19c200-0253-11ea-8687-48ce4066c113.PNG)

**Comment by MRayermannMSFT** [comment]
Can you clean out your GAC? We think that might be the issue. Here's a stack overflow post that may help you: https://stackoverflow.com/questions/33355114/how-to-remove-net-assembly-dlls-from-server-gac

**Comment by thereforeattila** [comment]
Hi,

Which one should I clean exactly? There are loads of stuff in the GAC folder, I don't want to delete everything from there. Could you please be a bit more specific in what I should try? Thank you.

**Comment by crmann1** [comment]
First lets check to see if the file is there.

If you browse to  C:\Windows\Microsoft.NET\assembly\GAC_MSIL   and look for a folder called Microsoft.Azure.Management.ResourceManager.   If it is there let us know. If it is not we will have to try another way to see where the assembly is loading from.

**Comment by thereforeattila** [comment]
Thanks for the reply. I checked and there is no such folder I am afraid.

**Comment by MRayermannMSFT** [comment]
@thereforeattila are you still having issues?

**Comment by thereforeattila** [comment]
> @thereforeattila are you still having issues?

Yeah, unfortunately it's still the same. Get the same error even after updating to the latest version now (1.12.0):

Unable to sign in: Method 'get_SerializationSettings' in type 'Microsoft.Azure.Management.ResourceManager.SubscriptionClient' from assembly 'Microsoft.Azure.Management.ResourceManager, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.

Any other ideas what I can try?

**Comment by thereforeattila** [comment]
For now as a workaround I am using storage account keys and passwors instead of my Azure account. But it's not too convenient.

**Comment by MRayermannMSFT** [comment]
@thereforeattila can you try the following for me?

1. Install process explorer if you do not already have it on your machine: https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer
2. Open process explorer
3. Open Storage Explorer
4. Try signing in
5. After it fails, go to process explorer
6. Look for the process `ServiceHub.Host.CLR.x86.exe
7. Right click on it and choose properties
8. Go to the .NET Assemblies tab
9. Look for our troublesome DLL and let us know its path
![image](https://user-images.githubusercontent.com/21112697/73561397-c6df2d80-440d-11ea-9dbd-0a549e400c07.png)

**Comment by MRayermannMSFT** [comment]
Closing due to lack of response. Please open a new issue if you're still having problems. Thanks.

**Comment by thereforeattila** [comment]
Hi, sorry about the late reply. the path  is the same as in your screenshot:

![image](https://user-images.githubusercontent.com/57538696/75249681-0283d200-57d7-11ea-81c3-2e7de2787b0e.png)

Hope this helps.

**Comment by thereforeattila** [comment]
Could we re-open this issue? I don't see why it makes sense to open a new issue with exactly the same problem.

Thanks!

**Comment by MRayermannMSFT** [comment]
@thereforeattila the reason we ask you to open a new issue is because in general, once an issue is closed, we don't look at it. We spend time every week going through open issues, making sure we're all staying on top of replying to y'all, brainstorming solutions as a team, discussing priority of issues, etc. And if an issue opener doesn't reply after 30 days, then we make the decision to assume that the issue is no longer a priority for the user and we close it. That way we can spend time on issues which are priorities for users. So next time, please open a new issue if you are still wanting help!

That said, I've always kept this issue in the back of my head, and when someone else opened a similar issue to this recently, I immediately remembered this one! So I'm going to reopen this one and direct them here. And since we're doing that, we'll also keep trying to help you if you're still needing help.

**Comment by MRayermannMSFT** [comment]
@RaijinDS please try each of the following:

Initial troubleshooting step:
> Can you clean out your GAC? We think that might be the issue. Here's a stack overflow post that may help you: https://stackoverflow.com/questions/33355114/how-to-remove-net-assembly-dlls-from-server-gac

And if that fails:
>     1. Install process explorer if you do not already have it on your machine: https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer
> 
>     2. Open process explorer
> 
>     3. Open Storage Explorer
> 
>     4. Try signing in
> 
>     5. After it fails, go to process explorer
> 
>     6. Look for the process `ServiceHub.Host.CLR.x86.exe
> 
>     7. Right click on it and choose properties
> 
>     8. Go to the .NET Assemblies tab
> 
>     9. Look for the troublesome DLL and let us know its path
 ![image](https://user-images.githubusercontent.com/21112697/73561397-c6df2d80-440d-11ea-9dbd-0a549e400c07.png)

**Comment by RaijinDS** [comment]
Hello Matthew, 

Thanks for your reply 👍 

Here the informations asked : 
![image](https://user-images.githubusercontent.com/57623002/96558760-6efa9680-12bc-11eb-848c-c61e5d3c632c.png)

> Microsoft.Azure.Management.ResourceManager	
> DomainNeutral	
> C:\Program Files (x86)\Microsoft Azure Storage Explorer\resources\app\ServiceHub\ADAL-Services\IdentityService\Microsoft.Azure.Management.ResourceManager.dll

See in attachment the list of the content of C:\Windows\GAC_MSIL.
Which assembly am I suppose to delete ? 

[list_gac_msil.txt](https://github.com/microsoft/AzureStorageExplorer/files/5407200/list_gac_msil.txt)

**Comment by MRayermannMSFT** [comment]
@RajinDS unfortunately the person who was guiding me, guide y'all, to figure out this problem, is on vacation till next week. I'm not sure what he was going to do with this information! I'll keep researching but we may have to wait for him to come back.

**Comment by RaijinDS** [comment]
> @RajinDS unfortunately the person who was guiding me, guide y'all, to figure out this problem, is on vacation till next week. I'm not sure what he was going to do with this information! I'll keep researching but we may have to wait for him to come back.

Hello @MRayermannMSFT, 

Don't worry, i'm on vacation too :) 
I can wait for its return ^^

**Comment by MRayermannMSFT** [comment]
@thereforeattila @RaijinDS so can you share with me the file at the path? 

It looks for both of y'all, that path is: `C:\Program Files (x86)\Microsoft Azure Storage Explorer\resources\app\ServiceHub\ADAL-Services\IdentityService\Microsoft.Azure.Management.ResourceManager.dll`

You can upload directly to this github issues by dragging and dropping the file into the comment text box.

**Comment by thereforeattila** [comment]
@MRayermannMSFT sure, please find it attached. Hope this helps 👍 
[Microsoft.Azure.Management.ResourceManager.dll.zip](https://github.com/microsoft/AzureStorageExplorer/files/5512624/Microsoft.Azure.Management.ResourceManager.dll.zip)

**Comment by RaijinDS** [comment]
> @thereforeattila @RaijinDS so can you share with me the file at the path?
> 
> It looks for both of y'all, that path is: `C:\Program Files (x86)\Microsoft Azure Storage Explorer\resources\app\ServiceHub\ADAL-Services\IdentityService\Microsoft.Azure.Management.ResourceManager.dll`
> 
> You can upload directly to this github issues by dragging and dropping the file into the comment text box.

Hello, 

Here my dll :)

[Microsoft.Azure.Management.ResourceManager.dll.zip](https://github.com/microsoft/AzureStorageExplorer/files/5515523/Microsoft.Azure.Management.ResourceManager.dll.zip)


Regards,

**Comment by MRayermannMSFT** [comment]
For both of you, the DLLs look fine. Do either of you use teams? We'd like to call you so we can debug this on your machines.

**Comment by MRayermannMSFT** [comment]
@thereforeattila @RaijinDS we have another idea! So back in process explorer, in the .NET assemblies for that ServiceHub process, look for Newtonsoft.JSON:
![image](https://user-images.githubusercontent.com/21112697/101966095-b110cc00-3bcb-11eb-9432-f6ee9326cc8f.png)
There will probably be two entries.

If that path for those are not in the Storage Explorer install directory, then:
1. Go to the Netwonsoft folder, for example, in my screenshot, it is `C:\Windows\assembly\NativeImages_v4.0.30319_32\Newtonsoft.Json`
2. In that folder, you'll see a bunch of folders like this:
![image](https://user-images.githubusercontent.com/21112697/101966281-fdf4a280-3bcb-11eb-8339-61079656d4d5.png)
3. Backup those folders somewhere else, like on your Desktop or something
4. Restart Storage Explorer and see if the error still happens

If it still happens then let me know so we can do some more brainstorming/try a Teams call. 😢

If it doesn't happen, then also let me know so we can finally close this issue! 🎉🙃

**Comment by thereforeattila** [comment]
@MRayermannMSFT thanks, I'll be able to check this in the next few days. Will get back to you as soon as I have the answer.

**Comment by MRayermannMSFT** [comment]
@thereforeattila @RaijinDS any updates?

**Comment by MRayermannMSFT** [comment]
It pains me to close this, but after 2 months of no replies I am going to. @thereforeattila @RaijinDS feel free to open a new issue if this is still happening, or comment here. I'll try to check on this issue every now and then.

**Comment by thereforeattila** [comment]
Hi @MRayermannMSFT 
Sorry about the late reply, finally checked this and it seems the issue is solved with removing all the folders from the "C:\Windows\assembly\NativeImages_v4.0.30319_32\Newtonsoft.Json" folder. Not sure if it is really this which solved the problem because I also updated to April 2021 (Version 1.19.0) at the same time I removed those folders (I know, should have done it one after the other :) ).

In any case, I hope this info helps others who may run into the problem.

Thanks a bunch for the help, much appreciated!
